command/hcl2_upgrade.go

```diff
@@ -326,33 +326,64 @@ func jsonBodyToHCL2Body(out *hclwrite.Body, kvs map[string]interface{}) {
 
 		switch value := value.(type) {
 		case map[string]interface{}:
-			var first interface{}
-			for _, elem := range value {
-				first = elem
+			var mostComplexElem interface{}
+			for _, randomElem := range value {
+				switch randomElem.(type) {
+				case string, int, float64, bool:
+					if mostComplexElem != nil {
+						continue
+					}
+					mostComplexElem = randomElem
+				default:
+					mostComplexElem = randomElem
```

Comment:
you could break the for loop here, once it finds something different from string, int, float64, bool. I found it hard to understand the logic, it seems that is more important to find the common types, but actually knowing that one different type exists here is enough to make the next decision, right?

Reply:
that is very true a break can be there; I'll adding it, hoping it makes reading better/easier

Reply:
ooohh! good to know! this is a detail that I wasn't remembering 🤔